### ISSUE TITLE
10$: Warn on ==, !, literals on left-hand side of ??

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#950]
The `??` operator does nothing useful if the left-hand side is never null. We can report a diagnostic in some easy cases.

Some examples of code which should warn:
```typescript
true ?? false
x == y ?? false
!b ?? false
"hi" ?? false
```

Inspiration: @Boshen's linter: https://twitter.com/boshen_c/status/1634813504204181504/photo/2

**Comment by applepie23** [comment]
Hi! I'm a student developer with the University of Michigan, and I, along with another student, are hoping to get involved in an open source project. We would love to take this issue!

**Comment by strager** [comment]
@applepie23 I assigned the task to you. Feel free to ask any questions if you need help.

**Comment by applepie23** [comment]
Hi, I tried to ask this question in Libera.Chat but I think my question disappeared. I am having a lot of trouble getting clang-format version 9.0.X to work. I was wondering if I could manually use clang-format version 16 and the quick-lint-provided .clang-format file to style all the files I edited instead of using the ./tools/clang-format executable. Thank you.

**Comment by strager** [comment]
> Hi, I tried to ask this question in Libera.Chat but I think my question disappeared.

I saw it. But you left within 5 minutes, so I couldn't respond.

> I was wondering if I could manually use clang-format version 16 and the quick-lint-provided .clang-format file to style all the files I edited instead of using the ./tools/clang-format executable.

No, please don't do that!

If you can't set up the right clang-format version, don't worry about formatting. I can format your patch before merging.

**Comment by strager** [comment]
Implemented in #969.
